lib/matplotlib/backend_bases.py

```diff
@@ -3195,3 +3207,560 @@ def zoom(self, *args):
+
+        self._tools[name] = tool_cls(self, name, *args, **kwargs)
+
+        if tool_cls.keymap is not None:
+            self.set_tool_keymap(name, tool_cls.keymap)
+
+        # For toggle tools init the radio_group in self._toggled
+        if isinstance(self._tools[name], tools.ToolToggleBase):
+            # None group is not mutually exclusive, a set is used to keep track
+            # of all toggled tools in this group
+            if tool_cls.radio_group is None:
+                self._toggled.setdefault(None, set())
+            else:
+                self._toggled.setdefault(tool_cls.radio_group, None)
+
+        self._tool_added_event(self._tools[name])
+        return self._tools[name]
+
+    def _tool_added_event(self, tool):
+        s = 'tool_added_event'
+        event = ToolEvent(s, self, tool)
+        self._callbacks.process(s, event)
+
+    def _handle_toggle(self, tool, sender, canvasevent, data):
+        """
+        Toggle tools, need to untoggle prior to using other Toggle tool
+        Called from tool_trigger_event
+        """
+
+        radio_group = tool.radio_group
+        # radio_group None is not mutually exclusive
+        # just keep track of toggled tools in this group
+        if radio_group is None:
+            if tool.toggled:
+                self._toggled[None].remove(tool.name)
+            else:
+                self._toggled[None].add(tool.name)
+            return
+
+        # If the tool already has a toggled state, untoggle it
+        if self._toggled[radio_group] == tool.name:
+            toggled = None
+        # If no tool was toggled in the radio_group
+        # toggle it
+        elif self._toggled.get(radio_group, None) is None:
+            toggled = tool.name
+        # Other tool in the radio_group is toggled
+        else:
+            # Untoggle previously toggled tool
+            self.tool_trigger_event(self._toggled[radio_group],
+                                    self,
+                                    canvasevent,
+                                    data)
+            toggled = tool.name
+
+        # Keep track of the toggled tool in the radio_group
+        self._toggled[radio_group] = toggled
+#         for a in self.canvas.figure.get_axes():
+#             a.set_navigate_mode(self._toggled)
+
+    def _get_cls_to_instantiate(self, callback_class):
+        # Find the class that corresponds to the tool
+        if isinstance(callback_class, six.string_types):
+            # FIXME: make more complete searching structure
+            if callback_class in globals():
+                return globals()[callback_class]
+
+            mod = self.__class__.__module__
+            current_module = __import__(mod,
+                                        globals(), locals(), [mod], 0)
+
+            return getattr(current_module, callback_class, False)
+
+        return callback_class
+
+    def tool_trigger_event(self, name, sender=None, canvasevent=None,
```

Comment:
This name is a bit too ambiguous, can it be `process_tool_trigger` or `emit_tool_trigger` or something like that?  (I don't have my head around the code enough to have a good idea yet, but I don't like _this_ name)